solutions/usaco-718.mdx

```diff
@@ -0,0 +1,92 @@
+---
+id: usaco-718
+source: USACO Gold 2017 February
+title: Why Did the Cow Cross The Road II
+author: Brad Ma
+---
+ 
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_nocross_gold_feb17.html)
+
+The normal recursive approach where you try all configurations of crosswalks has the 
+time complexity $\mathcal{O}(2 ^ N)$ which is extremely slow. We observe that we have 
+many duplicate recursive calls, so by using a top down dynamic progamming approach with 
+memoization we can significantly reduce the time complexity to $\mathcal{O}(N ^ 2)$.
+In this approach, we just store our answers so if there ever is a call to it again, the
+answer is just returned. A bottom up dynamic progamming approach without recursion can 
```

Comment:
is mentioning bottom up really necessary

Reply:
yes because you can do dp with recursion + memorization or dp without recursion